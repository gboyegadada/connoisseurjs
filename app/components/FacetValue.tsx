import React, { ReactNode, FC } from 'react'
import { FacetItem, FacetValue } from '../types/facet'
import { MdCheckBox, MdCheckBoxOutlineBlank } from 'react-icons/md';
import { createFacetConnect } from '../redux/store';

interface FacetValueProps {
    children?: ReactNode
    data: FacetValue,
    key: string
    field: string
}

const FacetConnect = createFacetConnect({
    mapActions: actions => ({
        tick: actions.tickFacetValue,
        untick: actions.untickFacetValue
    })
})

const FacetValue: FC<FacetValueProps> = (props) => {
    const { key, field, data: { value, numberOfResults, checked, exclude } } = props
    const numFormat = new Intl.NumberFormat('en-IN');

    const toggleTick = (a: any) => {
        checked ? a.untick({ field, value }) : a.tick({ field, value })
    }

    return (
        <FacetConnect>
        {(_, actions) => (
        <li className='facet-value' key={key} data-value={value} data-count={numberOfResults} onClick={() => toggleTick(actions)}>
            { checked ? <MdCheckBox size={21} /> : <MdCheckBoxOutlineBlank size={21} /> }
            
            <span className="facet-value-label">{value}</span> <span className="facet-value-count">{numFormat.format(numberOfResults)}</span>
        </li>
        )}
        </FacetConnect>
    )
}

export default FacetValue