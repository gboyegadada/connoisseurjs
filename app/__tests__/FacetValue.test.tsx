import React from 'react';
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import FacetValue from "../components/FacetValue";


describe("<FacetValue />", () => {
  test("should display clickable Facet Value", async () => {
    const main = shallow(<FacetValue 
            key='tpdisponibilite'
            field='tpdisponibilite'
            data={{ 
                value: 'En Succursale', 
                lookupValue: 'En Succursale', 
                score: 1, 
                valueType: 'Standard', 
                numberOfResults: 1, 
                checked: false, 
                exclude: false 
            }}
        />)

    expect(main).toMatchSnapshot()
  });
});