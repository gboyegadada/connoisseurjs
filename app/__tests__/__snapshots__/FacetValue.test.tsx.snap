// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FacetValue /> should display clickable Facet Value 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
